File: Eunleelee/2024-01/1월 6일/크레인_인형뽑기_게임.py
def solution(board, moves):
    answer = 0
    li = []
    num = -1
    for i in range(len(moves)):
        s = 0
        for j in range(len(board)):
            if board[j][moves[i]-1] != 0 and s == 0:
               li.append(board[j][moves[i]-1]) 
               board[j][moves[i]-1] = 0
               num += 1
               s = 1
        if num > 0: #num이 1부터
            if li[num] == li[num-1]:
                del li[-1]
                del li[-1]
                answer += 2
                num -= 2
    return answer

File: yht0827/1월/1월 6일/크레인 인형뽑기 게임.java
import java.util.Stack;

class Solution {
	public int solution(int[][] board, int[] moves) {
		int answer = 0;
		Stack<Integer> basket = new Stack<>();

		for (int i = 0; i < moves.length; i++) {
			for (int j = 0; j < board.length; j++) {
				if (board[j][moves[i] - 1] != 0) {

					if (!basket.isEmpty()) {
						if (basket.peek() == board[j][moves[i] - 1]) {
							basket.pop();
							board[j][moves[i] - 1] = 0;
							answer += 2;
							break;
						}
					}

					basket.push(board[j][moves[i] - 1]);
					board[j][moves[i] - 1] = 0;
					break;
				}
			}
		}

		return answer;
	}
}